sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -664,6 +665,29 @@ class CastSuite extends CastSuiteBase {
     }
   }
 
+  test("SPARK-35820: Support cast DayTimeIntervalType in different fields") {
+    val ymPositive = Literal.create(
+      Duration.ofSeconds(12345678L, 123456789),
+      DayTimeIntervalType(DAY, SECOND))
+    val ymNegative = Literal.create(
+      Duration.ofSeconds(-12345678L, -123456789),
```

Comment:
You can store `Duration.ofSeconds(12345678L, 123456789)` to a val, and call `.negated()`

Reply:
> You can store `Duration.ofSeconds(12345678L, 123456789)` to a val, and call `.negated()`

Done